SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc

```diff
@@ -0,0 +1,598 @@
+                              << "\n"
+                              << "Entry/Exit:"
+                              << "\n"
+                              << "Entry - X: " << hit.entryPoint().x() << " Y: " << hit.entryPoint().y()
+                              << " Z: " << hit.entryPoint().z() << "\n"
+                              << "Exit - X: " << hit.exitPoint().x() << " Y: " << hit.exitPoint().y()
+                              << " Z: " << hit.exitPoint().z() << "\n"
+
+                              << "Entry - X Pixel: " << hitEntryPointPixel.x() << " Y Pixel: " << hitEntryPointPixel.y()
+                              << "\n"
+                              << "Exit - X Pixel: " << hitExitPointPixel.x() << " Y Pixel: " << hitExitPointPixel.y()
+                              << "\n"
+
+                              << "row min: " << irow_min << " col min: " << icol_min << "\n";
+#endif
+
+  if (!(irow_min <= hitrow_max && irow_max >= hitrow_min && icol_min <= hitcol_max && icol_max >= hitcol_min)) {
+    // The clusters do not have an overlap, hence the hit is NOT reweighted
+    return false;
+  }
+
+  float cmToMicrons = 10000.f;
+
+  track[0] = (hitPosition.x() - origin.x()) * cmToMicrons;
+  track[1] = (hitPosition.y() - origin.y()) * cmToMicrons;
+  track[2] = 0.0f;  //Middle of sensor is origin for Z-axis
+  track[3] = direction.x();
+  track[4] = direction.y();
+  track[5] = direction.z();
+
+  array_2d pixrewgt(boost::extents[TXSIZE][TYSIZE]);
+
+  for (int row = 0; row < TXSIZE; ++row) {
+    for (int col = 0; col < TYSIZE; ++col) {
+      pixrewgt[row][col] = 0;
+    }
+  }
+
+  for (int row = 0; row < TXSIZE; ++row) {
+    xdouble[row] = topol->isItBigPixelInX(hitPixel.first + row - THX);
+  }
+
+  for (int col = 0; col < TYSIZE; ++col) {
+    ydouble[col] = topol->isItBigPixelInY(hitPixel.second + col - THY);
+  }
+
+  for (int row = 0; row < TXSIZE; ++row) {
+    for (int col = 0; col < TYSIZE; ++col) {
+      //Fill charges into 21x13 Pixel Array with hitPixel in centre
+      pixrewgt[row][col] =
+          hitSignal[PixelDigi::pixelToChannel(hitPixel.first + row - THX, hitPixel.second + col - THY)];
+    }
+  }
+
+  if (PrintClusters) {
+    std::cout << "Cluster before reweighting: " << std::endl;
+    printCluster(pixrewgt);
+  }
+
+  int ierr;
+  // for unirradiated: 2nd argument is IDden
+  // for irradiated: 2nd argument is IDnum
+  if (UseReweighting == true) {
+    int ID1 = dbobject_num->getTemplateID(detID);
+    int ID0 = dbobject_den->getTemplateID(detID);
+
+    if (ID0 == ID1) {
+      return false;
+    }
+    ierr = PixelTempRewgt2D(ID0, ID1, pixrewgt);
+  } else {
+    ierr = PixelTempRewgt2D(IDden, IDden, pixrewgt);
+  }
+  if (ierr != 0) {
+#ifdef TP_DEBUG
+    LogDebug("PixelDigitizer ") << "Cluster Charge Reweighting did not work properly.";
+#endif
+    return false;
+  }
+
+  if (PrintClusters) {
+    std::cout << "Cluster after reweighting: " << std::endl;
```

Comment:
use #ifdef TP_DEBUG or LogDEbug

Reply:
@civanch I disagree with this proposal. The purpose of the "PrintClusters" bool on line 303 (and 326 on your next comment) is to activate specifically this print out. By default this variable is set to false in the python, so there is no risk to get print out messages in standard running. It was coded like that by the original author when it was included in SiPixelDigitizerAlgorithm.cc. I prefer not to change.